### PR TITLE
Remove default limits

### PR DESCRIPTION
**Pull request by eric-sap**
## Proposed Changes

- 🎁 Allow not having the cpu/memory resource requests and/or limits in the config-kafka configmap for eventing-kafka.receiver and eventing-kafka.dispatcher
- 🧽 Default to no limit for receiver and dispatcher (but still allow a limit if defined by the admin in the config-kafka configmap) and a reasonable (100m / 50Mi) initial request size based on empirical small-cluster and small-message-size testing.

### CODE REVIEW

File: pkg/channel/distributed/controller/kafkachannel/reconciler_test.go [hunk 1]
```diff
@@ -191,6 +191,56 @@ func TestReconcile(t *testing.T) {
 			},
 		},
 
+		//
+		// Full Reconciliation (No Dispatcher Resource Requests Or Limits)
+		//
+
```
Comment (travis-minke-sap) [inline pkg/channel/distributed/controller/kafkachannel/reconciler_test.go:197]: This shouldn't be a new "section" of tests with the triple comment separator - it's just another Complete Reconciliation test instance - remove comment and change the two test names to be different.

File: pkg/channel/distributed/controller/kafkasecret/reconciler_test.go [hunk 2]
```diff
@@ -119,7 +119,34 @@ func TestReconcile(t *testing.T) {
...
+			WantPatches: []clientgotesting.PatchActionImpl{controllertesting.NewKafkaSecretFinalizerPatchActionImpl()},
+			WantEvents: []string{
+				controllertesting.NewKafkaSecretFinalizerUpdateEvent(),
+				controllertesting.NewKafkaSecretSuccessfulReconciliationEvent(),
+			},
+			OtherTestData: map[string]interface{}{
+				"configOptions": []controllertesting.KafkaConfigOption{controllertesting.WithNoReceiverResources},
+			},
+		},
 		//
```
Comment (travis-minke-sap) [inline pkg/channel/distributed/controller/kafkasecret/reconciler_test.go:150]: maintain linefeed separators (please ; )

File: pkg/channel/distributed/controller/kafkachannel/reconciler_test.go (hunk 1 above)
Comment (eric-sap) [reply pkg/channel/distributed/controller/kafkachannel/reconciler_test.go]: Done; not sure why I did that in the first place.

File: pkg/channel/distributed/controller/kafkasecret/reconciler_test.go (hunk 2 above)
Comment (eric-sap) [reply pkg/channel/distributed/controller/kafkasecret/reconciler_test.go:151]: certainly

**Pull request merged**